src/bigint.ts

```diff
@@ -0,0 +1,29 @@
+import type { Scalar, ScalarTag } from 'yaml'
+import { StringifyContext, stringifyString } from 'yaml/util'
+
+const identify = (value: any) => {
+  return typeof value === 'bigint' || value instanceof BigInt
+}
+
+/**
+ * `!bigint` BigInt
+ *
+ * [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) values,
+ * using their conventional `123n` representation.
+ */
+export const bigint = {
+  identify,
+  tag: '!bigint',
+  resolve(str: string) {
+    const match = str.match(/^([1-9][0-9]*|0x[0-9a-fA-F]+|0o[0-7]+|0b[01]+)n?$/)
```

Comment:
This is missing negative numbers and `0`. Tests should also cover those.

```suggestion
    const match = str.match(/^(-?(?:0|[1-9][0-9]*|0x[0-9a-fA-F]+|0o[0-7]+|0b[01]+))n?$/)
```

Reply:
The suggestion is a subset of that supported by JS, as `BigInt('0000001')` and `BigInt('-0')` are also valid. Not aware of a real reason why they should be supported though.

Reply:
I think it's fine for the YAML expression of BigInt to use canonical integer forms. Really, even having hex/octal/binary is just a flex 😅